_problems/channel-balance-probing.md

```diff
@@ -0,0 +1,120 @@
+
+To avoid getting probed and otherwise attacked, routing nodes may decide to only forward payments from trusted peers. This would be an undesirable development, as arguably the whole point of Lightning (and Bitcoin) is to be permissionless money. However, LN nodes have (relatively) persistent identities, and forwarding payments implies renting scarce resource (liquidity) to strangers. Therefore it's likely for trust-based countermeasures to emerge organically.
+
+
+### Deliberate balance sharing
+
+We may envision an additional protocol that would allow LN nodes willingly share their balance (up to some precision) to help others route payments through them. Such protocol may be seen as a "carrot" complementing "sticks" described above: nodes that want to give up privacy for more reliability may do so willingly and in a more elegant way than opening themselves up to probing.
+
+An additional line of research here could be using zero-knowledge proofs for privacy-preserving / verifiable balance sharing. One node would prove to the other that its balance lies in a certain range without revealing any more information.
+
+
+## Related Research
+
+1. Herrera-Joancomartí et al. in [*On the difficulty of hiding the balance of Lightning network channels*] introduced the general idea of probing in the LN.
+2. Van Dam in [*Improvements of the balance discovery attack on Lightning network payment channels*] suggested probing channels from both ends, which reduces the capital requirement for the attack.
+3. Kappos et al in [*An empirical analysis of privacy in the Lightning network*] proposed controlling both the sender and the receiver of probes. This removes the dependency of the attack on error messages being propagated back to the sender: the attacker always knows whether the probe reached the final destination.
+4. Tikhomirov et al. in [*Probing channel balances in the Lightning network*] implemented multi-hop probing. Sending probes along multi-hop paths allows for a) accumulating balance information about multiple channels at once; b) avoiding on-chain fees for opening channels to target hops.
+5. Biryukov et al. in [*Analysis and Probing of Parallel Channels in the Lightning Network*] explored probing of parallel channels. If the target hop contains parallel channels, the prober, in the general ase, doesn't know which of them forwarded the probe. This obstacle may be solved with jamming-enhanced probing, i.e., probing parallel channels one by one while blocking the remaining ones.
+
+
+## Commentary
+
+<!-- This is where you can post relevant informal and opinionated comments from various sources on the problem. -->
+<!-- Also you or anyone else can add conjecture to this section (after review). -->
+<!-- In general, this is not a comments section (use the issue for that). -->
+
+*There has been no commentary on this problem so far*
+
+
+## Related Problems
+
+- [Channel jamming attacks]
+- [Optimally reliable payment flows]
```

Comment:
The related problems section is links to other pages on the site so I'd remove these. I think you might want to add Rene's paper to the related research though since it analyses the efficiency of routing with probabilistic uncertainty around the balance.

The [removing cross layer links](https://bitcoinproblems.org/problems/removing-cross-layer-links.html) problem should be mentioned though. Effectively solving this would allow you to remove the concept of a "channel" altogether from the routing protocol. Nodes could instead just advertise a minimum capacity they will attempt to ensure they have between any node and use parallel channels + JIT re balancing etc to maintain them. From what I am hearing this may be the direction things are going in.

Reply:
Great vision on removing channel from routing, never thought about it this way! Added.